### PR TITLE
Improve object filtering

### PR DESCRIPTION
**Pull request by melegiul**
Thank you for sharing this awesome app, it does a great job :tada: 
This PR allows to pass a function in a linklist class to apply advanced filters like Subqueries, Q and F objects, etc
and solves #77 partially and can be combined with the current object filter options
Thanks in advance for your feedback

### CODE REVIEW

**Comment by claudep** [comment]
Thanks Giuliano for this proposal, I think it makes sense. Would it be possible to add a test for the new functionality? I'd also move the documentation from the code to the README (under Linklist classes).

**Comment by melegiul** [comment]
Thank you very much for your review.
I have added a test case for the filter_callable attribute and moved its documentation into the README.

File: README.rst [hunk 1]
```diff
@@ -83,6 +83,16 @@ customize the extracted links:
...
 
+    ``filter_callable``: a callable which allows to pass a function as filter
+    for your linklist class. It allows to apply more advanced filter operations.
+    This function must be a class method and it should be passed the objects query
+    set and return the filtered objects.
+    Example usage in your linklists.py - only check latest versions:
+    @classmethod
+    def filter_callable(cls, objects):
+        latest = Model.objects.filter(id=OuterRef('id')).order_by('-version')
+        return objects.filter(version=Subquery(latest.values('version')[:1]))
```
Comment (claudep) [inline README.rst]: For proper rest code syntax, I guess we need `::` and maybe indentation. Could you please try to improve this?

File: linkcheck/tests/sampleapp/linklists.py [hunk 2]
```diff
@@ -16,7 +17,18 @@ class AuthorLinklist(Linklist):
     url_fields = ['website']
 
 
+class JournalLinklist(Linklist):
+    """ Class to let linkcheck app discover fields containing links """
+    model = Journal
+    html_fields = ['description']
+    @classmethod
```
Comment (claudep) [inline linkcheck/tests/sampleapp/linklists.py:25]: Please add empty line above

**Review by claudep — COMMENTED** [review]
Almost ready :-)

File: README.rst (hunk 1 above)
Comment (melegiul) [reply README.rst]: You are absolutely right, I have fixed now the syntax

**Comment by claudep** [comment]
Thanks a lot for your contribution!

**Comment by melegiul** [comment]
@claudep Thank you once again, it was a real pleasure to work with you :slightly_smiling_face: 
Is there anything I can do to help releasing this one? We would be very happy to use this for our next migration

**Comment by claudep** [comment]
Sure, I'll prepare a new release soon.

**Comment by claudep** [comment]
Released 1.9.1

**Pull request merged**